### PR TITLE
ASUB-8195 Sign up with reCaptcha

### PR DESCRIPTION
**Pull request by LauraPinilla**
**Be sure to run `npm test`, `npm run lint`, and write detailed test steps before requesting reviewers**

## Description
This PR includes reCaptcha v2 & v3 for login & Sign Up
Login: https://arcpublishing.atlassian.net/browse/ASUB-8192
Sign Up: https://arcpublishing.atlassian.net/browse/ASUB-8195

The bot-challenge-protection component was moved into [arc-themes-component](ASUB-8193_8192_reCaptchav2_v3) repo, since this component should be shared across different blocks (identity & Subscriptions).

If reCaptcha is enabled the reCaptcha component should be appear

Screenshot 2024-04-03 at 8 26 33 p m

## Test Steps

_Add detailed test steps a reviewer must complete to test this PR_

1. Checkout this branch `git checkout ASUB-8195_SignUp_reCaptcha`
2. For arc-themes component checkout to this branch `git checkout ASUB-8193_8192_reCaptchav2_v3`
2. Run fusion repo with linked blocks `npx fusion start -l @wpmedia/identity-block`
3. Create one page for Login. Includes the "Identity Log In - Arc Block" and set up all the custom fields
4. Create one page for Sign Up. Includes the "Identity Sign Up - Arc Block" and set up all the custom fields
5. For both of them the reCaptcha v2 or v3 should be rendered if it is enabled for the tenant

`org: subssinglesite` `site:the-gazette` `env: sandbox` has recapctha v2 enabled for login & Sign up
         cdn: https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com
         
`org: subssinglesite` `site:the-daily-intelligencer` `env: sandbox` has recapctha v3 enabled for login & Sign up
         cdn: https://subssinglesite-the-daily-intelligencer-sandbox.api.cdn.arcpublishing.com

### CODE REVIEW

File: blocks/identity-block/features/signup/default.test.jsx [hunk 1]
```diff
@@ -21,83 +65,103 @@ describe("With initialized identity", () => {
...
+		await waitFor(() => expect(screen.getByRole("button")));
+		fireEvent.click(screen.getByRole("button"));
+	
+		await waitFor(() => {
+			expect(signUpMock).toHaveBeenCalled();
+		});
+	});
+
+	it("rejects the form", async () => {
+		const signUpMock = jest.fn(() => Promise.reject({code: 0})); // eslint-disable-line
```
Comment (malavikakoppula) [inline blocks/identity-block/features/signup/default.test.jsx]: @LauraPinilla valid rejection reason would fix the error on promise.

File: blocks/identity-block/features/social-sign-on/default.test.jsx [hunk 2]
```diff
@@ -21,20 +21,12 @@ describe("Subscriptions Social Login Feature", () => {
 		useIdentity.mockImplementation(() => ({ isInitialized: false }));
 
 		render(<SocialSignOnBlock customFields={defaultCustomFields} />);
-		expect(screen.queryByTestId("social-sign-on-container")).toBeNull();
+		expect(screen.queryByTestId("social-sign-on-container")).toBeNull(); // eslint-disable-line
```
Comment (malavikakoppula) [inline blocks/identity-block/features/social-sign-on/default.test.jsx]: @LauraPinilla This line doesn't have any lint error. // eslint-disable-line isn't needed in this line

File: blocks/identity-block/intl.json [hunk 3]
```diff
@@ -1303,5 +1305,11 @@
 	},
 	"identity-block.sign-up-natural": {
 		"en": "Sign up"
+	},
+	"identity-block.signup-form-error.identity-already-exists": {
+		"en": "Account already exists"
+	},
+	"identity-block.terms-service-privacy-text": {
```
Comment (malavikakoppula) [inline blocks/identity-block/intl.json]: @LauraPinilla Is there any reason why this is just added for "en"? Why it's different for other languages?

**Review by malavikakoppula — CHANGES_REQUESTED** [review]

File: blocks/identity-block/intl.json (hunk 3 above)
Comment (LauraPinilla) [reply blocks/identity-block/intl.json]: Hi @malavikakoppula Unfortunately, we don't have access to the themes lokalise account.

**Review by LauraPinilla — COMMENTED** [review]
@malavikakoppula. It's ready for review. Let's merge first one of the two PRs I have open, then I can solve the merge conflicts. And I can create the lokalize order from one of them.
Let me know if it works for you.

**Review by malavikakoppula — APPROVED** [review]
LGTM

**Pull request merged**